mscore/qml/palettes/PalettesWidget.qml

```diff
@@ -42,8 +42,6 @@ Item {
         paletteTree.applyCurrentElement();
     }
 
-    FocusChainBreak {}
```

Comment:
Without this it should probably be impossible to make focus go out of palettes with Tab key. Unfortunately it requires some hacks to make it possible to make focus leave [a window container](https://doc.qt.io/qt-5/qwidget.html#createWindowContainer) on some key press. Although this is not the only option to implement this, this special `FocusChainBreak` item seems to be the most reliable solution of those I have tried.

Reply:
@dmitrio95, this is deliberate. The idea is that you open the palette with F9, find your symbol, then close it again. There is no advantage to keeping the palette open for keyboard users, as they have to press a shortcut to focus it again anyway.

However, you raise a valid point which is how to escape the palette if you don't already know the F9 shortcut. I checked now, and pressing the Esc key returns focus to the score.

Reply:
BTW, as well as returning focus to the score, I tried to make the Esc key actually close the palette widget:

```c++
void PaletteWidget::keyPressEvent(QKeyEvent* event)
      {
      if (event->matches(QKeySequence::Cancel)) // Esc key or platform equivalent
            close();
      else
            QmlDockWidget::keyPressEvent(event);
      }
```

However, the key gets swallowed somewhere before it reaches this function and MuseScore's Escape shortcut is called instead. The same function worked on the Inspector's dock widget, so my guess is it gets swalled in the QML. Not sure how best to deal with this.

Anyway, focus is returned to the score on Esc, and that is the important thing for the time being.

Reply:
I just discovered something interesting / useful here.  I've long noticed that sometimes the screenreader would just stop responding, and never could figure out what triggered it.  Now I know for sure two things: applying a palette element, and hitting Esc.  Both return focus to the score as far as the keybaord is concerned, but somewhere the screenreader seems to not catch on to this.  It reads everything as expected when you first return to the score, but then subsequent navigation reads nothing.  My workaround if hit Alt to move fovcus to menu, then Esc.

I think what is happening is that somehow the accessibility system doesn't realize focus is now in the scoreview.  I think at one point I may have even seen something in the code to make me suspect this case was possible, but I didn't know how to trigger it.  But this is reproducible for me: screenreader stops responding after return from palette, whehter this happens by pressing Enter, Esc, or F9.

Reply:
I too am not really comfortable with having palettes close on hitting Enter, or Esc.  that would make sense if we otherwise treated the palettes like a modal dialog, but we don't. I suppose *that* could be the trigger: an "open palette as modal" command.  And blind users may indeed prefer to use that.  Then Enter or Esc could do what the currently do in the usual case, but could trigger the close in the modal case.

Reply:
@MarcSabatella, James at RNIB noticed that problem too. Thanks for the Alt workaround, I'll pass it on.

As I sighted user, I don't like how much space the various dock widgets take up on the screen, especially on a laptop or other small screen devices. I personally would prefer the palettes to close on Enter. I want to use them like Windows search:

1. Hit the Windows key to open the Start menu (or F9 to use the palette).
2. Type what you are looking for and find it in the results.
3. Hit Return to make your choice and dismiss the others.

I accept that not everybody uses the palettes in this way (perhaps the possibility never occured to them) but this should illustrate that there is no "correct" way to use the interface.

Reply:
I do like the idea behind the workflow you describe, and it does align with my idea for a "open as modal" command.  Not that I love the idea of an actual separate command, but somehow supporting both workflows.